docs/samples/Microsoft.ML.Samples/Dynamic/DataOperations/CrossValidationSplit.cs

```diff
@@ -0,0 +1,161 @@
+            PrintPreviewRows(folds[0]);
+
+            // The data in the Train split.
+            // [Group, 1], [Features, 0.8173254]
+            // [Group, 2], [Features, 0.7680227]
+            // [Group, 1], [Features, 0.2060332]
+            // [Group, 2], [Features, 0.5588848]
+            // [Group, 1], [Features, 0.4421779]
+            // [Group, 2], [Features, 0.9775497]
+            // 
+            // The data in the Test split.
+            // [Group, 0], [Features, 0.7262433]
+            // [Group, 0], [Features, 0.5581612]
+            // [Group, 0], [Features, 0.9060271]
+            // [Group, 0], [Features, 0.2737045]
+            PrintPreviewRows(folds[1]);
+            // The data in the Train split.
+            // [Group, 0], [Features, 0.7262433]
+            // [Group, 2], [Features, 0.7680227]
+            // [Group, 0], [Features, 0.5581612]
+            // [Group, 2], [Features, 0.5588848]
+            // [Group, 0], [Features, 0.9060271]
+            // [Group, 2], [Features, 0.9775497]
+            // [Group, 0], [Features, 0.2737045]
+            // 
+            // The data in the Test split.
+            // [Group, 1], [Features, 0.8173254]
+            // [Group, 1], [Features, 0.2060332]
+            // [Group, 1], [Features, 0.4421779]
+            PrintPreviewRows(folds[2]);
+            // The data in the Train split.
+            // [Group, 0], [Features, 0.7262433]
+            // [Group, 1], [Features, 0.8173254]
+            // [Group, 2], [Features, 0.7680227]
+            // [Group, 0], [Features, 0.5581612]
+            // [Group, 1], [Features, 0.2060332]
+            // [Group, 2], [Features, 0.5588848]
+            // [Group, 0], [Features, 0.9060271]
+            // [Group, 1], [Features, 0.4421779]
+            // [Group, 2], [Features, 0.9775497]
+            // [Group, 0], [Features, 0.2737045]
+            // 
+            // The data in the Test split.
+            // [Group, 2], [Features, 0.7680227]
+            // [Group, 2], [Features, 0.5588848]
+            // [Group, 2], [Features, 0.9775497]
+
+            // Example of a split without specifying a sampling key column.
+            folds = mlContext.Data.CrossValidationSplit(dataview, numberOfFolds: 3);
+            PrintPreviewRows(folds[0]);
+
+            // The data in the Train split.
+            // [Group, 0], [Features, 0.7262433]
+            // [Group, 1], [Features, 0.8173254]
+            // [Group, 2], [Features, 0.7680227]
+            // [Group, 0], [Features, 0.5581612]
+            // [Group, 1], [Features, 0.2060332]
+            // [Group, 1], [Features, 0.4421779]
+            // [Group, 2], [Features, 0.9775497]
+            // [Group, 0], [Features, 0.2737045]
+            // 
+            // The data in the Test split.
+            // [Group, 2], [Features, 0.5588848]
+            // [Group, 0], [Features, 0.9060271]
```

Comment:
This also does not look good. Can you check this as well?

Reply:
Why?
Data got broken into 4/4/2 chunks.
You not happy with distribution?

---
In reply to: [268366691](https://github.com/dotnet/machinelearning/pull/3058#discussion_r268366691) [](ancestors = 268366691)

Reply:
OK, that's TrainTestSplit in each fold are not expected to be of same size right?

---
In reply to: [268367175](https://github.com/dotnet/machinelearning/pull/3058#discussion_r268367175) [](ancestors = 268367175,268366691)